packages/storage/src/providers/s3/apis/internal/copy.ts

```diff
@@ -97,6 +98,18 @@ const copyWithPath = async (
 		destination,
 		identityId,
 	);
+	const { expectedSourceBucketOwner, expectedBucketOwner } = {
+		...(source.expectedBucketOwner && {
+			expectedSourceBucketOwner: validateBucketOwnerID(
+				source.expectedBucketOwner,
+			)?.accountID,
+		}),
+		...(destination.expectedBucketOwner && {
+			expectedBucketOwner: validateBucketOwnerID(
+				destination.expectedBucketOwner,
+			)?.accountID,
+		}),
+	};
```

Comment:
I'm not sure what the spread and assign operator achieves here. Why not 
```suggestion
	validateBucketOwnerID(source.expectedBucketOwner);
	validateBucketOwnerID(
		destination.expectedBucketOwner,
	);
```